scripts/table_updates/update_data_table.py

```diff
@@ -423,79 +454,258 @@ def update_redact_table(
     syn.store(Table(full_pt_schema, result, etag=pt_dat_query.etag))
 
 
-def custom_fix_for_cancer_panel_test_table(
+def check_if_all_join_keys_available(
+    left_table: pandas.DataFrame,
+    right_table: pandas.DataFrame,
+    left_on: str,
+    right_on: str,
+    logger: logging.Logger = None,
+):
+    """Check if all join keys are available in the right table
+
+    Args:
+        left_table (pandas.DataFrame): The left table
+        right_table (pandas.DataFrame): The right table
+        left_on (str): The join key of the left table
+        right_on (str): The join key of the right table
+        logger (logging.Logger, optional): The custom logger. Optional. Defaults to None.
+    Returns:
+        logging.Logger: warning message if join key is not available in the right table
+    """
+    missing_join_keys = left_table.loc[
+        ~left_table[left_on].isin(right_table[right_on])
+    ][left_on]
+    if len(missing_join_keys) > 0:
+        logger.warning(
```

Comment:
Hmm have you run into this before where main genie patient ids may not match BPC patient ids?

Reply:
Nope. I added this to address Chelsea's and your question: "Do we know what happens if a main GENIE value is missing? Or if the sample/patient is missing?" + "what happens if there isn't a 1:1 merge between bpc and main genie (BPC has sample/patients not present in clinical)".

Reply:
I think warning might makes the most sense in this context since we just wanted to be informed that some of the tier1a fields will be filled with NaN.